main/src/main/scala/sbt/internal/RemoteCache.scala

```diff
@@ -41,12 +45,30 @@ object RemoteCache {
       .map(_.take(commitLength))
 
   lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
-    remoteCacheId := gitCommitId,
-    remoteCacheIdCandidates := gitCommitIds(5),
+    remoteCacheId := "",
+    remoteCacheIdCandidates := Nil,
     pushRemoteCacheTo :== None
   )
 
   lazy val projectSettings: Seq[Def.Setting[_]] = (Seq(
+    remoteCacheId := {
+      val compileExtraInc = (Compile / extraIncOptions).value
+      val compileInputs = (Compile / unmanagedSources / inputFileStamps).value
+      val compileCp = (Compile / externalDependencyClasspath / outputFileStamps).value
```

Comment:
Could this be defined at the configuration level (and combined at the project level via a scope filter), so that other configurations such as IntegrationTest can more easily be supported? https://github.com/sbt/sbt/blob/5a960f2f94b8c5260e07e380b76e88f1d6959cd4/main/src/main/scala/sbt/internal/RemoteCache.scala#L206-L207 would still need to be exposed, maybe within the same `Seq[Def.Setting[_]]` ?

Reply:
Good idea. I'm hoping I can eventually implement `pullRemoteCache` at the configuration level.